### PR TITLE
NF: Add new scripts in scilpy public

### PR DESCRIPTION
**Pull request by GuillaumeTh**
Add scil_add_tracking_mask_to_pft_maps.py, scil_convert_sh_basis.py, scil_flip_gradients.py

### CODE REVIEW

File: scripts/scil_convert_sh_basis.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    assert_inputs_exist(parser, [args.input_sh])
+    assert_outputs_exists(parser, args, [args.output_name])
+
+    if args.sh_basis == 'tournier07':
```
Comment (jchoude) [inline scripts/scil_convert_sh_basis.py]: The `if` could be simplified to
```
input_basis = args.sh_basis
output_basis = 'descoteaux07' if input_basis == 'tournier07' else 'tournier07'
```

File: scripts/scil_flip_gradients.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
+#! /usr/bin/env python
+
```
Comment (jchoude) [inline scripts/scil_flip_gradients.py]: You don't have the coding line.

File: scripts/scil_flip_gradients.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                   help='Path to encoding file.')
+
+    p.add_argument('flipped_encoding', action='store',
+                   metavar='flipped_encoding', type=str,
+                   help='Path to the flipped encoding file.')
+
+    p.add_argument('axes', action='store', metavar='dimension',
+                   choices=['x', 'y', 'z'], nargs='+',
+                   help='The axes you want to flip. eg: to flip the x '
+                        ' and y axes use: x y')
```
Comment (jchoude) [inline scripts/scil_flip_gradients.py]: One space too many in the help string.

File: scripts/scil_flip_gradients.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+from scilpy.io.utils import (assert_inputs_exist, assert_outputs_exists)
+from scilpy.utils.flip_tools import flip_mrtrix_encoding_scheme, flip_fsl_bvecs
+from scilpy.utils.util import str_to_index
+
+
+def build_arg_parser():
+    p = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter,
+                                description='Flip one or more axes of the '
+                                            ' encoding scheme matrix.')
```
Comment (jchoude) [inline scripts/scil_flip_gradients.py]: One space too many.

File: scripts/scil_flip_gradients.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+    p.add_argument('encoding_file', action='store', metavar='encoding_file',
```
Comment (jchoude) [inline scripts/scil_flip_gradients.py]: No need for `action='store'`, `type` and `metavar` when you use the defaults.

File: scripts/scil_flip_gradients.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+                   type=str,
+                   help='Path to encoding file.')
+
+    p.add_argument('flipped_encoding', action='store',
```
Comment (jchoude) [inline scripts/scil_flip_gradients.py]: Same for unused.

File: scripts/scil_flip_gradients.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+                   metavar='flipped_encoding', type=str,
+                   help='Path to the flipped encoding file.')
+
+    p.add_argument('axes', action='store', metavar='dimension',
```
Comment (jchoude) [inline scripts/scil_flip_gradients.py]: Same for unused.

File: scripts/scil_flip_gradients.py [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+                        ' and y axes use: x y')
+
+    gradients_type = p.add_mutually_exclusive_group(required=True)
+    gradients_type.add_argument('--fsl', dest='fsl_bvecs', action='store_true',
+                                help='Specify fsl format')
+    gradients_type.add_argument('--mrtrix', dest='fsl_bvecs',
+                                action='store_false',
+                                help='Specify mrtrix format')
+
+    p.add_argument('-f', action='store_true', dest='overwrite',
```
Comment (jchoude) [inline scripts/scil_flip_gradients.py]: You could use the centralized functions.

File: scripts/scil_flip_gradients.py [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+import argparse
+
+from scilpy.io.utils import (add_overwrite_arg, assert_inputs_exist,
+                             assert_outputs_exists)
+from scilpy.utils.flip_tools import flip_mrtrix_encoding_scheme, flip_fsl_bvecs
+from scilpy.utils.util import str_to_index
+
+
+def build_arg_parser():
```
Comment (GuillaumeTh) [inline scripts/scil_flip_gradients.py]: add _ at the beginning

File: scripts/scil_convert_sh_basis.py [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+def _build_arg_parser():
+    p = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter,
+                                description=DESCRIPTION)
+
+    p.add_argument('input_sh',
+                   help='Input SH filename (nii or nii.gz)')
+
+    p.add_argument('output_name',
+                   help='Name of the output file')
+
```
Comment (arnaudbore) [inline scripts/scil_convert_sh_basis.py]: Add a . for each help sentence

**Pull request merged**